feat(oxc_cfg):  better control flow graph dot dot repr

IWANABETHATGUY opened the pull request:
1. add basic block index for each basicblock, so there should no basic block display empty string
2. use `shape` box, since it is basic **block**
## Before
![image](https://github.com/user-attachments/assets/9b556cca-8401-40bd-b3b6-d31c5e7cf8ba)

## After
![image](https://github.com/user-attachments/assets/b8e4c52d-ecdf-4e92-85be-ea37f0cb2330)

rzvxa reviewed (approved):
This is really nice😍

rzvxa commented:
On another kind of related note; It will be super cool if we add 2 checkboxes to the playground for adding some extra information to the graph and show/hide implicit error paths.

I sometimes use this example to get extra debug information while I'm tracking something. It's a bit noisy but it has helped me quite a few times and might be useful to others as well.

https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/examples/cfg.rs

IWANABETHATGUY committed:
feat(oxc_cfg):  better control flow graph dot dot repr (#5731)

1. add basic block index for each basicblock, so there should no basic block display empty string
2. use `shape` box, since it is basic **block**
## Before
![image](https://github.com/user-attachments/assets/9b556cca-8401-40bd-b3b6-d31c5e7cf8ba)

## After
![image](https://github.com/user-attachments/assets/b8e4c52d-ecdf-4e92-85be-ea37f0cb2330)

IWANABETHATGUY commented:
> On another kind of related note; It will be super cool if we add 2 checkboxes to the playground for adding some extra information to the graph and show/hide implicit error paths.
> 
> I sometimes use this example to get extra debug information while I'm tracking something. It's a bit noisy but it has helped me quite a few times and might be useful to others as well.
> 
> https://github.com/oxc-project/oxc/blob/main/crates/oxc_semantic/examples/cfg.rs

Sure, I will have a look.
